packages/repository/src/__tests__/fixtures/mixins/find-by-title-repo-mixin.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Constructor} from '@loopback/context';
+import {CrudRepository, Entity, Where} from '../../..';
+
+export interface FindByTitle<E extends Entity> {
+  findByTitle(title: string): Promise<E[]>;
+}
+
+/*
+ * This function adds a new method 'findByTitle' to a repository class
+ * where 'E' is a model which extends Entity
+ *
+ * @typeParam E - Model class which extends Entity
+ */
+
+export function FindByTitleRepositoryMixin<
+  E extends Entity & {title: string},
+  R extends Constructor<CrudRepository<E>>
```

Comment:
It seems to me that you are mixing two clean approaches here and arriving at a mixed and a kind of "dirty" solution.

- Approach 1: Require only Model + CRUD methods: use `M extends Model & {title: string}` and `R extends Constructor<CrudRepository<M>>`
- Approach 2: Require `Entity` + CRUD + entity-specific methods, to make the mixin easier to use: use `E extends Entity & {title: string}` and `R extends Constructor<EntityCrudRepository<E>>`.

IIRC, the idea behind the second approach was to simplify definition of `NoteRepository` so that it does not have to specify full repository type as you have now:

```ts
export class NoteRepository extends FindByTitleRepositoryMixin<
   Note,
   Constructor<
     DefaultCrudRepository<Note, typeof Note.prototype.id, NoteRelations>
   >
 >(DefaultCrudRepository)  {
  // ...
}
```

IIRC, we were aiming for the following usage in Approach 2:

```ts
export class NoteRepository extends FindByTitleRepositoryMixin<
   Note,
   typeof Note.prototype.id,
   NoteRelations,
 >(DefaultCrudRepository) {
  // ...
}
```

Based on @raymondfeng proposal in https://github.com/strongloop/loopback-next/pull/4778, I think we should simply revert back to Approach 1 and use `Model` + `CrudRepository` in the generic arguments of the mixin.

Reply:
@bajtos, @raymondfeng 

I cannot seem to make this work:

> R extends Constructor<EntityCrudRepository<E, ID, Relations>>

VSCode complains about ID and Relations.

Reply:
> It seems to me that you are mixing two clean approaches here and arriving at a mixed and a kind of "dirty" solution. 

Not sure why using Entity instead of Model is a dirty approach. Can you clarify?

Reply:
With the introduction of `R extends Constructor<CrudRepository<M>>` in your PR, I was able to downgrade `Entity` to `Model` in #4778 and all tests are passing. I think we now have a good solution for the repo mixin to use `Model` directly.